.github/workflows/ci.yaml

```diff
@@ -130,11 +130,65 @@ jobs:
           name: coverage-${{ matrix.python-version }}
           path: .coverage
 
+  tests-linux-alpha:
+    name: tests / run / ${{ matrix.python-version }} / Linux
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        python-version: ["3.11-dev"]
+    outputs:
+      python-key: ${{ steps.generate-python-key.outputs.key }}
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v3.0.1
+        with:
+          fetch-depth: 0
+      - name: Set up Python ${{ matrix.python-version }}
+        id: python
+        uses: actions/setup-python@v3.1.2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Generate partial Python venv restore key
+        id: generate-python-key
+        run: >-
+          echo "::set-output name=key::venv-${{ env.CACHE_VERSION }}-${{
+            hashFiles('setup.cfg', 'requirements_test.txt', 'requirements_test_min.txt',
+          'requirements_test_brain.txt') }}"
+      - name: Restore Python virtual environment
+        id: cache-venv
+        uses: actions/cache@v3.0.2
+        with:
+          path: venv
+          key: >-
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{
+            steps.generate-python-key.outputs.key }}
+          restore-keys: |
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-venv-${{ env.CACHE_VERSION }}-
+      - name: Create Python virtual environment
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+        run: |
+          python -m venv venv
+          . venv/bin/activate
+          python -m pip install -U pip setuptools wheel
+          pip install -U -r requirements_test.txt -r requirements_test_brain.txt
+          pip install -e .
+      - name: Run pytest
+        run: |
+          . venv/bin/activate
+          pytest --cov --cov-report= tests/
+      - name: Upload coverage artifact
+        uses: actions/upload-artifact@v3.0.0
+        with:
+          name: coverage-${{ matrix.python-version }}
+          path: .coverage
+
   coverage:
     name: tests / process / coverage
     runs-on: ubuntu-latest
     timeout-minutes: 5
-    needs: ["tests-linux"]
+    needs: ["tests-linux", "tests-linux-alpha"]
+    if: ${{ always() }}
```

Comment:
@cdce8p This remains necessary for the coverage to include `3.11` which would be good for any changes we introduce to fix regression with that version. I have removed it from the windows job though.

Reply:
Not sure that is all that useful. The coverage is merged anyway, so the only thing we would get would be the lines with `PY311_PLUS`.

Reply:
I think that would still be helpful. For example, more "risky" PRs such as https://github.com/PyCQA/astroid/pull/1513 would probably be less risky if we know coverage is okay.